src/objective/multiclass_objective.hpp

```diff
@@ -46,11 +60,12 @@ class MulticlassLogloss: public ObjectiveFunction {
           auto p = rec[k];
           size_t idx = static_cast<size_t>(num_data_) * k + i;
           if (label_int_[i] == k) {
-            gradients[idx] = static_cast<score_t>(p - 1.0f);
+            gradients[idx] = static_cast<score_t>(p - 1.0f) * label_pos_weights_[k];
+            hessians[idx] = static_cast<score_t>(2.0f * p * (1.0f - p))* label_pos_weights_[k];
           } else {
             gradients[idx] = static_cast<score_t>(p);
+            hessians[idx] = static_cast<score_t>(2.0f * p * (1.0f - p));
```

Comment:
@cbecker 
I think your idea is also a solution. But i am not sure which one will be better. 
My idea is to multiply a big weight to gradient/hessian with small-frequency class, so that the model will give bigger prediction to them.
Your idea is to give sample weights for different class, and softmax will give bigger prediction for these larger-weight data.